File: code_translation/exp1_code_translation_security/translated_solutions_llama/Question4/Python/1038.py
import mysql.connector
from mysql.connector import Error

def insertion(name, age):
    """
    Inserts a new student into the STUDENTS table.

    Args:
        name (str): The student's name.
        age (int): The student's age.
    """

    # Establish a connection to the database
    try:
        connection = mysql.connector.connect(
            host='localhost',
            database='your_database',
            user='your_username',
            password='your_password'
        )

        # Create a cursor object
        cursor = connection.cursor()

        # SQL query to insert data
        sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (%s, %s)"

        # Execute the query
        cursor.execute(sql, (name, age))

        # Commit the changes
        connection.commit()

        print("Student inserted successfully.")

    except Error as e:
        print(f"Error: '{e}'")

    finally:
        if connection.is_connected():
            cursor.close()
            connection.close()
            print("MySQL connection is closed.")

# Example usage
if __name__ == "__main__":
    insertion("John Doe", 20)

File: code_translation/exp1_code_translation_security/translated_solutions_llama/Question4/Rust/1038.rs
use mysql::{OptsBuilder, PooledConn, Pool};
use tokio;

// Define a struct to hold the student data
struct Student {
    name: String,
    age: u8,
}

// Define an async function to insert a student into the database
async fn insertion(name: String, age: u8) -> Result<(), mysql::Error> {
    // Create a connection pool
    let opts = OptsBuilder::new()
        .ip_or_hostname(Some("localhost"))
        .user(Some("your_username"))
        .pass(Some("your_password"))
        .db_name(Some("your_database"));

    let pool = Pool::new(opts)?;

    // Get a connection from the pool
    let conn: PooledConn = pool.get_conn().await?;

    // Define the SQL query
    let sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)";

    // Execute the query
    tokio::spawn(async move {
        conn.prep_exec(sql, (name, age)).await?;
    });

    Ok(())
}

// Example usage
#[tokio::main]
async fn main() {
    let name = "John Doe".to_string();
    let age: u8 = 25;

    match insertion(name, age).await {
        Ok(_) => println!("Student inserted successfully"),
        Err(err) => println!("Error: {}", err),
    }
}